sim/rogue/items_sets_pve.go

```diff
@@ -440,4 +440,108 @@ var ItemSetDeathdealersThrill = core.NewItemSet(core.ItemSet{
 			})
 		},
 	},
-})
\ No newline at end of file
+})
+
+var ItemSetDeathdealersBattlearmor = core.NewItemSet(core.ItemSet{
+	Name: "Deathdealer's Battlearmor",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Main Gauche now strikes 1 additional nearby target and also causes your Sinister Strike to strike 1 additional nearby target. 
+		// These additional strikes are not duplicated by Blade Flurry.
+		2: func(agent core.Agent) {
+			//Implemented in sinister_strike.go and main_gauche.go
+			rogue := agent.(RogueAgent).GetRogue()
+			if !rogue.HasRune(proto.RogueRune_RuneMainGauche) {
+				return
+			}
+			
+			var curDmg float64
+
+			cleaveHit := rogue.RegisterSpell(core.SpellConfig{
+				ActionID:    core.ActionID{SpellID: 1213754},
+				SpellSchool: core.SpellSchoolPhysical,
+				ProcMask:    core.ProcMaskEmpty,
+				Flags:       core.SpellFlagMeleeMetrics | core.SpellFlagNoOnCastComplete,
+		
+				DamageMultiplier: 1,
+				ThreatMultiplier: 1,
+		
+				ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+					spell.CalcAndDealDamage(sim, target, curDmg, spell.OutcomeAlwaysHit)
+				},
+			})	
+			
+			cleaveAura := rogue.RegisterAura(core.Aura{
+				Label:    "2P Cleave Buff",
+				Duration: time.Second * 10,
+				OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if rogue.Env.GetNumTargets() > 1 && result.Landed() && (spell.SpellCode == SpellCode_RogueSinisterStrike) {
```

Comment:
Instead of checking target counts within these callbacks, we know that this set is functionally useless on 1 target, so can we just not build all of these auras if `rogue.Env.GetNumTarget() == 1`? Basically just adding it to the guard at the top of the set bonus. That should be a nice performance improvement